geth/jail/cell.go

```diff
@@ -65,9 +67,30 @@ func (c *Cell) Stop() {
 // event queue loop and schedules for immediate execution.
 // Intended to be used by any cell user that want's to run
 // async call, like callback.
-func (c *Cell) CallAsync(fn otto.Value, args ...interface{}) {
+func (c *Cell) CallAsync(fn otto.Value, args ...interface{}) error {
 	task := looptask.NewCallTask(fn, args...)
-	c.lo.Add(task)
-	// TODO(divan): review API of `loop` package, it's contrintuitive
-	go c.lo.Ready(task)
+	errChan := make(chan error, 1)
+	doneChan := make(chan struct{})
```

Comment:
I'm not sure but it looks like goroutine leak. If there is no errors all looks fine - https://play.golang.org/p/iXlXinaRxG. But in case of any error https://play.golang.org/p/fejNf9AaKU : for-select cyrcle has been already closed and `<-doneChan` will never happend if so `doneChan <- struct{}{}` in goroutine will block forever.

Reply:
Very interesting. I have learned a lot about concurrency, specifically in Go while working on this. It seems that changing `doneChan` to a buffered channel fixes the issues that you have pointed out.

The solution I have come up with is changing to `doneChan := make(chan struct{}, 1)`

Reply:
It's good and correct solution!